### DISCUSSION TITLE
Provide a helper function to install all copyright/license to a directory

### DISCUSSION THREAD

**Discussion by reitowo** [microsoft/vcpkg#28980] [Ideas]
I've been using this to do so, hope vcpkg could provide a better helper function.
```cmake
# install licenses
file(GLOB LICENSE_DIRS "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/share/*")
foreach(LICENSE_DIR ${LICENSE_DIRS})
    if (EXISTS ${LICENSE_DIR}/copyright)
        get_filename_component(LICENSE_LIB_NAME ${LICENSE_DIR} NAME)
        message(STATUS ${LICENSE_LIB_NAME})
        install(FILES ${LICENSE_DIR}/copyright DESTINATION "license" RENAME ${LICENSE_LIB_NAME})
    endif()
endforeach()
```

**Comment by dg0yt** [comment]
I assume this is for your user project, not for a regular port. For a regular port, I don't see a use case for installing multiple files. (vcpkg wants a single copyright file.) For user projects, this needs to come in another way (CMake package?) than a port helper funtion.

**Reply by reitowo** [reply]
This is a situation that user might want to distribute all open source code licenses along with its distribution. So the above code basically just copy all copyright files and gather them in one destination

**Reply by dg0yt** [reply]
So this is really for the users' projects. I just wanted to clarify that this cannot be a port function like `vcpkg_install_copyright`.
Such a function could be provided via a CMake package. But it should be used in manifest mode only in order to reliably reflect the dependencies.

**Reply by reitowo** [reply]
Yes, that's what I mean
